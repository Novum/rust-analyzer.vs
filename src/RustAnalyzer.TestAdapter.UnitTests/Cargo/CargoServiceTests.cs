using System.IO;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using KS.RustAnalyzer.TestAdapter.Cargo;
using KS.RustAnalyzer.TestAdapter.Common;
using KS.RustAnalyzer.Tests.Common;
using Newtonsoft.Json;
using Xunit;

namespace KS.RustAnalyzer.TestAdapter.UnitTests.Cargo;

// TODO: MS: check for all crate-types in workspace_mixed.
public sealed class CargoServiceTests
{
    [Theory]
    [InlineData(@"workspace_mixed")]
    [UseReporter(typeof(DiffReporter))]
    public async Task GetMetadataTestsAsync(string workspaceRelRoot)
    {
        var workspaceRoot = (PathEx)Path.Combine(TestHelpers.ThisTestRoot, workspaceRelRoot);

        var wmd = await new CargoService(TestHelpers.TL.T, TestHelpers.TL.L).GetMetadata(workspaceRoot, default);

        var normalizedStr = wmd
            .SerializeObject(Formatting.Indented, new PathExJsonConverter())
            .Replace(TestHelpers.ThisTestRoot.Replace("\\", "\\\\"), "<TestRoot>");
        Approvals.Verify(normalizedStr);
    }
}
