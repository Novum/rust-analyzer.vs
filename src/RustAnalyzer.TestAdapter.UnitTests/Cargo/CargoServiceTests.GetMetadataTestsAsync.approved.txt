{
  "version": 1,
  "target_directory": "<TestRoot>\\workspace_mixed\\target",
  "workspace_root": "<TestRoot>\\workspace_mixed",
  "packages": [
    {
      "name": "shared",
      "manifest_path": "<TestRoot>\\workspace_mixed\\shared\\Cargo.toml",
      "targets": [
        {
          "name": "shared",
          "src_path": "<TestRoot>\\workspace_mixed\\shared\\src\\lib.rs",
          "kind": [
            "lib"
          ],
          "crate_types": [
            "lib"
          ]
        },
        {
          "name": "example1",
          "src_path": "<TestRoot>\\workspace_mixed\\shared\\src\\lib.rs",
          "kind": [
            "example"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "bench1",
          "src_path": "<TestRoot>\\workspace_mixed\\shared\\src\\lib.rs",
          "kind": [
            "bench"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "bench2",
          "src_path": "<TestRoot>\\workspace_mixed\\shared\\src\\lib.rs",
          "kind": [
            "bench"
          ],
          "crate_types": [
            "bin"
          ]
        }
      ]
    },
    {
      "name": "main",
      "manifest_path": "<TestRoot>\\workspace_mixed\\Cargo.toml",
      "targets": [
        {
          "name": "main",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\libx.rs",
          "kind": [
            "lib"
          ],
          "crate_types": [
            "lib"
          ]
        },
        {
          "name": "bin1",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "bin"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "bin2",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "bin"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "example1",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "example"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "example2",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "example"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "example3",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "example"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "test1",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "test"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "test2",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "test"
          ],
          "crate_types": [
            "bin"
          ]
        },
        {
          "name": "bench1",
          "src_path": "<TestRoot>\\workspace_mixed\\src\\main.rs",
          "kind": [
            "bench"
          ],
          "crate_types": [
            "bin"
          ]
        }
      ]
    }
  ]
}